### ISSUE TITLE
Handle variable attributes in the Foreign Interface, if possible

### ISSUE THREAD

**Issue by turibe** [SWI-Prolog/swipl-devel#1223]
It seems that the Foreign Interface does not have methods for creating attribute variables, and for extracting attributes from variables.

**Comment by JanWielemaker** [comment]
Indeed.   I've not come across a good use case.    If you think you have one, please discuss.   Please do so on the Discourse forum.

**Comment by JanWielemaker** [comment]
> looking for how to construct a rational number using the (C) Foreign Language interface

Use PL_put_term_from_chars().  That function is optimized for parsing numbers and is suitable for passing in big integers and rational numbers.  For big integers you best use hexadecimal syntax.   As is, rational numbers always use decimal syntax.

Otherwise closing as stale.   Please discuss on Discourse if you think you have a good reason to create attributes from C.    It is not very complicates, but adding an API that is never used is a waste of resources.
